setup.py

```diff
@@ -13,20 +13,26 @@
 requires = [r.strip() for r in req_data.split() if r.strip() != '']
 requires = list(reversed(requires))
 
+# testing dependencies
+req_data = open(os.path.join(__here__, 'test_requirements.txt')).read()
+tests_require = [r.strip() for r in req_data.split() if r.strip() != '']
+tests_require = list(reversed(tests_require))
+
 # temporary workaround for a Python 2 CherryPy bug, for which we opened a pull request:
 # https://bitbucket.org/cherrypy/cherrypy/pull-request/85/1285-python-2-now-accepts-both-bytestrings/
 if sys.version_info[0] == 2:
     requires = ['CherryPy==3.2.2' if 'cherrypy' in r.lower() else r for r in requires]
 
 if __name__ == '__main__':
+    setup_requires = {'setup_requires': ['distribute']} if sys.version_info[0] == 2 else {}
```

Comment:
Fact: packaging is the worst thing about Python.

setuptools was obsoleted by distribute which got merged back into setuptools, which means that ``setuptools`` both obsoletes and is obsoleted by ``distribute``.

Those of us using Python 3 such as MAGFest and new projects where I work can just ignore distribute, but older Python 2.7 projects still need distribute, so I changed this to only rely on distribute if you're on Python 2.7.

I don't think this should cause any problems for MAGFest, but it'll be something to keep an eye on.